File: CSE305/achrysan_HW2/Java/hw2.java
//package hw2;

import java.io.IOException;
import java.util.Iterator;
import java.util.Stack;
import java.io.*;

public class hw2 {
	
	private static Stack myStack = new Stack();
	public static void hw2(String input, String output){
		try {
			
		BufferedWriter write = new BufferedWriter(new FileWriter(output));
		BufferedReader read = new BufferedReader(new FileReader(input));
		String line;
		while((line = read.readLine()) != null){
			line = line.replace("\n","");
			System.out.println("The line is: "+line);
			if(line.length()>3 && !line.equals(":true:") && !line.equals(":false:") && !line.equals(":error:")  && !line.equals("swap") && !line.equals("quit")){
			if(line.substring(0, 4).equals("push")){
				try{
				myStack.push(Integer.parseInt(line.substring(5, line.length())));
				System.out.println(line.substring(5, line.length())+" is being pushed onto the stack.");
				System.out.println("The Stack now contains:");
				System.out.print(myStack.toString().replaceAll("\\[", "").replaceAll("]", "")+"\n"+"\n");
				}catch(Exception e){
					myStack.push(":error:");
					System.out.print(myStack.toString().replaceAll("\\[", "").replaceAll("]", "")+"\n"+"\n");
				}
			}else{ myStack.push(":error:");
			System.out.print(myStack.toString().replaceAll("\\[", "").replaceAll("]", "")+"\n"+"\n");}
			}else if(line.equals("pop")){
				try{
				myStack.pop();
				System.out.println("An element is being popped off!");
				}catch(Exception e){
					myStack.push(":error:");
				}
			}else if(line.equals(":true:") ||line.equals(":false:")||line.equals(":error:")){
				myStack.push(line);
				System.out.println(line +" is being pushed onto the stack.");
				System.out.println("The Stack now contains:");
				System.out.print(myStack.toString().replaceAll("\\[", "").replaceAll("]", "")+"\n"+"\n");
			}else if(line.equals("add")){
				add_vals();
			}else if(line.equals("sub")){
				sub_vals();
			}else if(line.equals("mul")){
				mul_vals();
			}else if(line.equals("div")){
				div_vals();
			}else if(line.equals("rem")){
				rem_vals();
			}else if(line.equals("neg")){
				neg_vals();
			}else if(line.equals("swap")){
				swap_vals();
			}else if(line.equals("quit")){
				System.out.println("This is what is being printed to the File: ");
				System.out.print(myStack.toString().replaceAll("\\[", "").replaceAll("]", "")+"\n"+"\n");
				Iterator<Object> iter = myStack.iterator();
				for(int i = myStack.size()-1; i>=0; i--){
					write.write(myStack.get(i).toString());
					write.newLine();
				}
				//System.exit(0);
				
			}else{
				myStack.push(":error:");
				}
				System.out.print(myStack.toString().replaceAll("\\[", "").replaceAll("]", "")+"\n"+"\n");
			}
		
		read.close();
		write.close();
	} catch (IOException e) {
		System.out.println("Error Occured:\n");
		e.printStackTrace();
	}
		
	}

	private static void swap_vals() {
		System.out.println("swapping");
		if(myStack.size()>1){
			Object temp = myStack.pop();
			Object temp2 = myStack.pop();
			myStack.push(temp);
			myStack.push(temp2);
			System.out.print(myStack.toString().replaceAll("\\[", "").replaceAll("]", "")+"\n"+"\n");
			}else{
				myStack.push(":error:");
				System.out.print(myStack.toString().replaceAll("\\[", "").replaceAll("]", "")+"\n"+"\n");
			}
		
	}

	private static void neg_vals() {
		System.out.println("negating");
		if(myStack.size()>0){
			Object myVal_1 = myStack.peek();
			if(myVal_1 instanceof Integer){
				Object myVal1 = myStack.pop();
			int	result = 0 - ((int)myVal1);
			System.out.println("result: "+result);
			myStack.push(result);
			System.out.print(myStack.toString().replaceAll("\\[", "").replaceAll("]", "")+"\n"+"\n");
			}else{
				myStack.push(":error:");
				System.out.print(myStack.toString().replaceAll("\\[", "").replaceAll("]", "")+"\n"+"\n");
			}
			}else{
				myStack.push(":error:");
				System.out.print(myStack.toString().replaceAll("\\[", "").replaceAll("]", "")+"\n"+"\n");
			}
		
		
	}

	private static void rem_vals() {
		System.out.println("modulating");
		if(myStack.size()>1){
			Object myVal_1 = myStack.peek();
			Object temp = myStack.pop();
			Object myVal_2 = myStack.peek();
			myStack.push(temp);
			if(myVal_1 instanceof Integer && myVal_2 instanceof Integer){
				Object myVal1 = myStack.pop();
				Object myVal2 = myStack.pop();
			int	result = ((int) myVal2) % ((int)myVal1);
			System.out.println("result: "+result);
			myStack.push(result);
			System.out.print(myStack.toString().replaceAll("\\[", "").replaceAll("]", "")+"\n"+"\n");
			}else{
				myStack.push(":error:");
				System.out.print(myStack.toString().replaceAll("\\[", "").replaceAll("]", "")+"\n"+"\n");
			}
			}else{
				myStack.push(":error:");
				System.out.print(myStack.toString().replaceAll("\\[", "").replaceAll("]", "")+"\n"+"\n");
			}
		
		
	}

	private static void div_vals() {
		System.out.println("dividing");
		if(myStack.size()>1){
			Object myVal_1 = myStack.peek();
			Object temp = myStack.pop();
			Object myVal_2 = myStack.peek();
			myStack.push(temp);
			if(myVal_1 instanceof Integer && myVal_2 instanceof Integer){
				Object myVal1 = myStack.pop();
				Object myVal2 = myStack.pop();
			int	result = ((int) myVal2) / ((int)myVal1);
			System.out.println("result: "+result);
			myStack.push(result);
			System.out.print(myStack.toString().replaceAll("\\[", "").replaceAll("]", "")+"\n"+"\n");
			}else{
				myStack.push(":error:");
				System.out.print(myStack.toString().replaceAll("\\[", "").replaceAll("]", "")+"\n"+"\n");
			}
			}else{
				myStack.push(":error:");
				System.out.print(myStack.toString().replaceAll("\\[", "").replaceAll("]", "")+"\n"+"\n");
			}
		
		
	}

	private static void mul_vals() {
		System.out.println("multiplying");
		if(myStack.size()>1){
			Object myVal_1 = myStack.peek();
			Object temp = myStack.pop();
			Object myVal_2 = myStack.peek();
			myStack.push(temp);
			if(myVal_1 instanceof Integer && myVal_2 instanceof Integer){
				Object myVal1 = myStack.pop();
				Object myVal2 = myStack.pop();
			int	result = ((int) myVal2) * ((int)myVal1);
			System.out.println("result: "+result);
			myStack.push(result);
			System.out.print(myStack.toString().replaceAll("\\[", "").replaceAll("]", "")+"\n"+"\n");
			}else{
				myStack.push(":error:");
				System.out.print(myStack.toString().replaceAll("\\[", "").replaceAll("]", "")+"\n"+"\n");
			}
			}else{
				myStack.push(":error:");
				System.out.print(myStack.toString().replaceAll("\\[", "").replaceAll("]", "")+"\n"+"\n");
			}
		
		
	}

	private static void sub_vals() {
		System.out.println("subtracting");
		if(myStack.size()>1){
			Object myVal_1 = myStack.peek();
			Object temp = myStack.pop();
			Object myVal_2 = myStack.peek();
			myStack.push(temp);
			System.out.println("myVal_1: "+myVal_1);
			System.out.println("myVal_2: "+myVal_2);
			if(myVal_1 instanceof Integer && myVal_2 instanceof Integer){
				Object myVal1 = myStack.pop();
				Object myVal2 = myStack.pop();
			int	result = ((int) myVal2) - ((int)myVal1);
			System.out.println("result: "+result);
			myStack.push(result);
			System.out.print(myStack.toString().replaceAll("\\[", "").replaceAll("]", "")+"\n"+"\n");
			}else{
				myStack.push(":error:");
				System.out.print(myStack.toString().replaceAll("\\[", "").replaceAll("]", "")+"\n"+"\n");
			}
			}else{
				myStack.push(":error:");
				System.out.print(myStack.toString().replaceAll("\\[", "").replaceAll("]", "")+"\n"+"\n");
			}
		
		
	}

	private static void add_vals() {
		System.out.println("adding");
		if(myStack.size()>1){
			Object myVal_1 = myStack.peek();
			Object temp = myStack.pop();
			Object myVal_2 = myStack.peek();
			myStack.push(temp);
		if(myVal_1 instanceof Integer && myVal_2 instanceof Integer){
			Object myVal1 = myStack.pop();
			Object myVal2 = myStack.pop();
		int	result = ((int) myVal2) + ((int)myVal1);
		System.out.println("result: "+result);
		myStack.push(result);
		System.out.print(myStack.toString().replaceAll("\\[", "").replaceAll("]", "")+"\n"+"\n");
		}else{
			myStack.push(":error:");
			System.out.print(myStack.toString().replaceAll("\\[", "").replaceAll("]", "")+"\n"+"\n");
		}
		}else{
			myStack.push(":error:");
			System.out.print(myStack.toString().replaceAll("\\[", "").replaceAll("]", "")+"\n"+"\n");
		}
		
	}

}

File: CSE305/achrysan_HW2/Python/hw2.py
#!/usr/bin/env python
import sys
myStack = list();

def hw2(inString, outString):
    file_read = open(inString, 'r')
    text_in_f = file_read.readlines()
    print(text_in_f)
    file_write = open(outString, 'w')
    for line in text_in_f:
        line = line[:-1]
        print("The line: \""+line+"\"")
        if line[:4] == "push":
            value = line[5:len(line)]
            try:
                if '.' in line:
                    push_val(":error:")
                else:
                    push_val(int(value))
            except:
                push_val(":error:")

        elif line == "pop":
            pop_val()#line[4:len(line)])
        elif line == ":true:" or line == ":false:" or line == ":error:":
            push_val(line)
        elif line == "add":
            add_vals()
        elif line == "sub":
            sub_vals()
        elif line == "mul":
            mul_vals()
        elif line == "div":
            div_vals()
        elif line == "rem":
            rem_vals()
        elif line == "neg":
            neg_val()
        elif line == "swap":
            swap_vals()
        elif line == "qui" or line == "quit":
            print("\nThis is what's being written to the file:")
            print(*myStack, sep="\n")
            for item in myStack:
                file_write.write("%s\n" % item)
            sys.exit()
            
def push_val(myVal):
    """if((not(isinstance(myVal,int))) and myVal!= ":true:" or myVal!= ":false:" or myVal != ":error:"):
        myStack.insert(0, ":error:")
        return myStack
    if(isinstance(myVal,int):
       myStack.insert("""
    myStack.insert(0,myVal)
    print(myVal,"is being pushed onto the stack")
    print(*myStack, sep="\n")
    print("")
    return myStack

def pop_val():
    if len(myStack) != 0:
        print(myStack[0],"is being pop'd off the stack")
        myStack.remove(myStack[0])
        return myStack
    else:
        pass

def add_vals():
    print("adding")
    if len(myStack)>1:
        myVal_1 = myStack[0]
        myVal_2 = myStack[1]
        if isinstance(myVal_1,int) and isinstance(myVal_2,int):
            result = myVal_2 + myVal_1
            print("result: ",result)
            pop_val()
            pop_val()
            push_val(result)
        else:
            push_val(":error:")
    else:
        push_val(":error:")


def sub_vals():
    print("subtracting")
    if len(myStack)>1:
        myVal_1 = myStack[0]
        myVal_2 = myStack[1]
        if isinstance(myVal_1,int) and isinstance(myVal_2,int):
            result = myVal_2 - myVal_1
            print("result: ",result)
            pop_val()
            pop_val()
            push_val(result)
        else:
            push_val(":error:")
    else:
        push_val(":error:")

def mul_vals():
    print("multiplying")
    if len(myStack)>1:
        myVal_1 = myStack[0]
        myVal_2 = myStack[1]
        if isinstance(myVal_1,int) and isinstance(myVal_2,int):
            result = myVal_2 * myVal_1
            print("result: ",result)
            pop_val()
            pop_val()
            push_val(result)
        else:
            push_val(":error:")
    else:
        push_val(":error:")

def div_vals():
    print("dividing")
    if len(myStack)>1:
        myVal_1 = myStack[0]
        myVal_2 = myStack[1]
        if isinstance(myVal_1,int) and isinstance(myVal_2,int) and myVal_2 != 0:
            result = myVal_2 // myVal_1
            print("result: ",result)
            pop_val()
            pop_val()
            push_val(result)
        else:
            push_val(":error:")
    else:
        push_val(":error:")

def rem_vals():
    print("modulating")
    if len(myStack)>1:
        myVal_1 = myStack[0]
        myVal_2 = myStack[1]
        if isinstance(myVal_1,int) and isinstance(myVal_2,int) and myVal_2 != 0:
            result = myVal_2 % myVal_1
            print("result: ",result)
            pop_val()
            pop_val()
            push_val(result)
        else:
            push_val(":error:")
    else:
        push_val(":error:")

def neg_val():
    print("negating")
    if len(myStack)>0:
        myVal_1 = myStack[0]
        if isinstance(myVal_1,int):
            result = 0 - myVal_1
            pop_val()
            push_val(result)
            print("result: ",result)
        else:
            push_val(":error:")
    else:
        push_val(":error:")

def swap_vals():
    print("swapping")
    if len(myStack)>1:
        temp = myStack[0]
        myStack[0] = myStack[1]
        myStack[1] = temp
        print(*myStack, sep="\n")
    else:
        push_val(":error:")
        print(*myStack, sep="\n")
